src/components/Dropdown/DropdownWithIcon.jsx

```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { settings } from '../../constants/Settings';
+
+import { Dropdown } from './index';
+
+const { iotPrefix } = settings;
+
+const propTypes = {
+  itemToString: PropTypes.func,
+  items: PropTypes.arrayOf(
+    PropTypes.shape({
+      id: PropTypes.string.isRequired,
+      text: PropTypes.string.isRequired,
+      icon: PropTypes.oneOfType([PropTypes.func, PropTypes.object]).isRequired,
+    })
+  ).isRequired,
+};
+
+const defaultPropTypes = {
+  itemToString: null,
+};
+
+const DropdownWithIcon = ({
+  selectedViewId,
+  items,
+  itemToString,
+  actions: { onChangeView, ...otherActions },
+  ...other
+}) => {
+  const onSelectionChange = (changes) => {
+    const { selectedItem } = changes;
+
+    onChangeView(selectedItem);
```

Comment:
Is this perhaps a leftover from the TableViewDropdown?

Reply:
I need to pull the selected item outside of the changes from here; i moved it down further since I no longer need it in that exact spot.